models/ssh_key.go

```diff
@@ -465,6 +466,29 @@ func SearchPublicKeyByContent(content string) (*PublicKey, error) {
 	return key, nil
 }
 
+// SearchPublicKey returns a list of public keys matching the provided arguments.
+func SearchPublicKey(uid int64, fingerprint string, keyType string, keyContent string) ([]*PublicKey, error) {
+	keys := make([]*PublicKey, 0, 5)
+	cond := builder.NewCond()
+	if uid != 0 {
+		cond = cond.And(builder.Eq{"owner_id": uid})
+	}
+	if fingerprint != "" {
+		cond = cond.And(builder.Eq{"fingerprint": fingerprint})
+	}
+	if keyContent != "" {
+		replacer := strings.NewReplacer("%", "%%")
```

Comment:
I think this is oracle syntax for escaping percent sign.
I think final qerry condition should be (this should work at least in sqlite, mysql, postgres and mssql):
`col like '%5\%%' escape '\'`
Through I don't know if xorm does support this (@lunny maybe you know?)

Reply:
Xorm doesn't support this, but we can use cond.Expr instead.  If you're sure that the ESCAPE keyword is allowed then I'll put that in. 

Is there a better way of doing these tests than using LIKE?

Reply:
OK, 4-5 attempts later I think I have a sensible working escaping function and working LIKE expression...